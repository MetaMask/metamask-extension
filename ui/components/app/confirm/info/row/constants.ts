export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';

export enum RowAlertKey {
  EstimatedFee = 'estimatedFee',
  SigningInWith = 'signingInWith',
  RequestFrom = 'requestFrom',
  Resimulation = 'resimulation',
  Speed = 'speed',
}

export enum AlertActionKey {
  Buy = 'buy',
  ShowAdvancedGasFeeModal = 'showAdvancedGasModal',
  ShowGasFeeModal = 'showGasFeeModal',
}
