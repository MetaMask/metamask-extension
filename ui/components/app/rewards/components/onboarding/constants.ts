export const REWARDS_ONBOARD_OPTIN_LEGAL_LEARN_MORE_URL =
  'https://support.metamask.io/manage-crypto/metamask-rewards';

export const REWARDS_ONBOARD_TERMS_URL = 'https://go.metamask.io/rewards-terms';
