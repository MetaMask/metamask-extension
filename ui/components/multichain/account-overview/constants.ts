export const FUND_SLIDE = {
  id: 'fund',
  title: 'slideFundWalletTitle',
  description: 'slideFundWalletDescription',
  image: './images/slide-fund-icon.svg',
  href: 'https://portfolio.metamask.io/buy/build-quote',
};

///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
export const BRIDGE_SLIDE = {
  id: 'bridge',
  title: 'slideBridgeTitle',
  description: 'slideBridgeDescription',
  image: './images/slide-bridge-icon.svg',
};
///: END:ONLY_INCLUDE_IF

export const CARD_SLIDE = {
  id: 'card',
  title: 'slideDebitCardTitle',
  description: 'slideDebitCardDescription',
  image: './images/slide-card-icon.svg',
  href: 'https://portfolio.metamask.io/card',
};

export const CASH_SLIDE = {
  id: 'cash',
  title: 'slideCashOutTitle',
  description: 'slideCashOutDescription',
  image: './images/slide-sell-icon.svg',
  href: 'https://portfolio.metamask.io/sell',
};

export const ZERO_BALANCE = '0x00';
