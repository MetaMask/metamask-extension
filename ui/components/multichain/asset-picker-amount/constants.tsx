export const LARGE_SYMBOL_LENGTH = 7;
