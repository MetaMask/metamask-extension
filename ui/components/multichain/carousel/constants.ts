export const MARGIN_VALUES = {
  SLIDE_BOTTOM: '40px',
  CONTAINER_SIDE: '16px',
  SLIDE_GAP: '2%',
  ZERO: '0',
};

export const WIDTH_VALUES = {
  FULL_WIDTH: '100%',
  STANDARD_SLIDE: '98%',
  FIRST_SLIDE: '94%',
};

export const BANNER_STYLES = {
  HEIGHT: '59px',
};

export const MAX_SLIDES = 5;
