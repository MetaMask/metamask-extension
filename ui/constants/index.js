export const ETH_DEFAULT_DECIMALS = 8;
