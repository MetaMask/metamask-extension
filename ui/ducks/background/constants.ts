import type {
  BackgroundStateProxy,
  MemStoreControllersComposedState,
} from '../../../shared/types/background';

export const initialBackgroundState: Omit<
  BackgroundStateProxy,
  keyof MemStoreControllersComposedState
> &
  Partial<{
    [ControllerName in keyof MemStoreControllersComposedState]: Partial<
      MemStoreControllersComposedState[ControllerName]
    >;
  }> = {
  isInitialized: false,
} as const;
