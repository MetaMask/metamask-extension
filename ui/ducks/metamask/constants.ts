import { NameType } from '@metamask/name-controller';
// TODO: Replace with import from `@metamask/preferences-controller` once migration to core repo is complete.
// eslint-disable-next-line import/no-restricted-paths
import { getDefaultPreferencesControllerState } from '../../../app/scripts/controllers/preferences-controller';
import type {
  BackgroundStateProxy,
  MemStoreControllersComposedState,
} from '../../../shared/types/metamask';
import {
  DEFAULT_AUTO_LOCK_TIME_LIMIT,
  ThemeType,
} from '../../../shared/constants/preferences';
import { LedgerTransportTypes } from '../../../shared/constants/hardware-wallets';

export const initialMetamaskState: Omit<
  BackgroundStateProxy,
  keyof MemStoreControllersComposedState
> &
  Partial<{
    [ControllerName in keyof MemStoreControllersComposedState]: Partial<
      MemStoreControllersComposedState[ControllerName]
    >;
  }> = {
  isInitialized: false,
  KeyringController: {
    isUnlocked: false,
    keyrings: [],
  },
  AccountsController: {
    internalAccounts: { accounts: {}, selectedAccount: '' },
  },
  AccountTracker: {
    currentBlockGasLimit: '',
    currentBlockGasLimitByChainId: {},
  },
  AddressBookController: {
    addressBook: {},
  },
  ApprovalController: {
    pendingApprovals: {},
    approvalFlows: [],
  },
  CurrencyController: {
    currentCurrency: 'usd',
    currencyRates: {
      ETH: {
        conversionRate: null,
        conversionDate: 0,
        usdConversionRate: null,
      },
    },
  },
  ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
  CustodyController: {
    custodyAccountDetails: {},
  },
  ///: END:ONLY_INCLUDE_IF
  DecryptMessageController: {
    unapprovedDecryptMsgs: {},
  },
  GasFeeController: {
    gasFeeEstimates: {},
    gasEstimateType: 'none',
  },
  MetaMetricsController: {
    participateInMetaMetrics: null,
    dataCollectionForMarketing: null,
  },
  NameController: {
    names: {
      [NameType.ETHEREUM_ADDRESS]: {},
    },
  },
  NetworkController: {
    selectedNetworkClientId: '',
    networkConfigurationsByChainId: {},
    networksMetadata: {},
  },
  NftController: {
    allNfts: {},
  },
  OnboardingController: {
    firstTimeFlowType: null,
    completedOnboarding: false,
  },
  PreferencesController: {
    ...getDefaultPreferencesControllerState(),
    theme: ThemeType.os,
    useBlockie: false,
    use4ByteResolution: true,
    useNftDetection: true,
    useTokenDetection: true,
    openSeaEnabled: true,
    securityAlertsEnabled: true,
    featureFlags: {},
    currentLocale: '',
    knownMethodData: {},
    // Ledger transport type is deprecated. We currently only support webhid
    // on chrome, and u2f on firefox.
    ledgerTransportType: window.navigator.hid
      ? LedgerTransportTypes.webhid
      : LedgerTransportTypes.u2f,
    preferences: {
      ...getDefaultPreferencesControllerState().preferences,
      autoLockTimeLimit: DEFAULT_AUTO_LOCK_TIME_LIMIT,
      showExtensionInFullSizeView: false,
      showFiatInTestnets: false,
      showNativeTokenAsMainBalance: true,
      showTestNetworks: false,
      smartTransactionsOptInStatus: true,
      petnamesEnabled: true,
      featureNotificationsEnabled: false,
      privacyMode: false,
      showMultiRpcModal: false,
      tokenSortConfig: {
        key: 'tokenFiatAmount',
        order: 'dsc',
        sortCallback: 'stringNumeric',
      },
      tokenNetworkFilter: {},
    },
  },
  SignatureController: {
    unapprovedPersonalMsgs: {},
    unapprovedTypedMessages: {},
  },
  TokensController: {
    allTokens: {},
  },
  TxController: {
    transactions: [],
  },
} as const;
