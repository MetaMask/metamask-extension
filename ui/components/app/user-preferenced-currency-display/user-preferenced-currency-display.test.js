import React from 'react';
import configureMockStore from 'redux-mock-store';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import { mockNetworkState } from '../../../../test/stub/networks';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import UserPreferencedCurrencyDisplay from '.';

describe('UserPreferencedCurrencyDisplay Component', () => {
  describe('rendering', () => {
    const defaultState = {
      metamask: {
        ...mockState.metamask,
        ...mockNetworkState({ chainId: CHAIN_IDS.MAINNET }),
        currencyRates: {},
        preferences: {},
      },
    };
    const mockStore = configureMockStore()(defaultState);
    it('should match snapshot', () => {
      const { container } = renderWithProvider(
        <UserPreferencedCurrencyDisplay />,
        mockStore,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
