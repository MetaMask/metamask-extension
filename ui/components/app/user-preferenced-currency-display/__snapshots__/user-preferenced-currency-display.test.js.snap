// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferencedCurrencyDisplay Component rendering should match snapshot 1`] = `
<div>
  <div
    class="mm-box currency-display-component mm-box--display-flex mm-box--flex-wrap-wrap mm-box--align-items-center"
    title="0 ETH"
  >
    <span
      class="mm-box mm-text currency-display-component__text mm-text--inherit mm-text--ellipsis mm-box--color-text-default"
      data-testid="account-value-and-suffix"
    >
      0
    </span>
    <span
      class="mm-box mm-text currency-display-component__suffix mm-text--inherit mm-box--margin-inline-start-1 mm-box--color-text-default"
    >
      ETH
    </span>
  </div>
</div>
`;
